lib/commands/context.js

```diff
@@ -192,8 +194,11 @@ async function setupExistingChromedriver (chromedriver) {
 }
 
 async function setupNewChromedriver (opts, curDeviceId, adbPort) {
+  let getPort = BlueBird.promisify(PortFinder.getPort, {context: PortFinder});
+  let chromePort = await getPort();
+  logger.error(`Using ${chromePort} for new ChromeDriver session`);
   let chromeArgs = {
-    port: opts.chromeDriverPort,
+    port: chromePort,
```

Comment:
I don't think we should be overwriting `opts.chromeDriverPort` if it is set. Only if it's unset. Or, we need to do the random port-finding in a higher-level package (like Android Driver)